app/models/user.rb

```diff
@@ -43,7 +45,16 @@ def minor
   end
 
   def insufficient_credit
-    provider == 'amber_oauth2' and credit < -5
+    provider == 'amber_oauth2' and credit.negative?
+  end
+
+  def can_order(activity = nil)
+    activity ||= current_activity
+    if activity.nil?
+      !insufficient_credit
+    else
+      !insufficient_credit or activity.orders.select { |order| order.user_id == id }.any?
+    end
```

Comment:
can the activity ever be nil?

Reply:
It shouldn't, but better be safe than sorry :)

Reply:
does ruby have short circuiting conditions? like && in js and java? then you could write something to the extent of 
return !insuff or (!activity.nil? && ...)